src/pdf/generator.py

```
@@ -0,0 +1,130 @@
+            # Set metadata
+            merged_pdf.docinfo["/Title"] = title
+            merged_pdf.docinfo["/Creator"] = "kindle2pdf"
+            
+            # Save
+            merged_pdf.save(output_path, linearize=True)
+        
+        logger.info(f"Merged {len(pdf_paths)} PDFs into: {output_path}")
+        return output_path
+        
+    except Exception as e:
+        raise PDFError(f"Failed to merge PDFs: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)